src/app/components/MusicKitAuthorizeProvider.jsx

```diff
@@ -57,14 +58,18 @@ class MusicKitAuthorizeProvider extends React.Component {
 
     if (!isAuthorized && !this.state.browsing) {
       return (
-        <Login
+        <SplashScreen
           onClick={() => this.props.mk.instance.authorize()}
           onBrowse={() => this.setState({ browsing: true })}
         />
       );
     }
 
-    return this.props.children;
+    return (
+      <AuthorizeContext.Provider value={isAuthorized}>
```

Comment:
The provider should be in App

Reply:
Putting the provider in the app would mean adding the event listener into the App. As this component was already listening for the authorization did change event, I think its best to keep the provider in this component.